modules/social_features/social_group/modules/social_group_request/social_group_request.module

```diff
@@ -0,0 +1,168 @@
+    !$group->hasField('field_group_allow_request')
+  ) {
+    $variables['allow_request'] = FALSE;
+    return;
+  }
+
+  $allow_request = $group->get('field_group_allow_request');
+  if ($allow_request->isEmpty() || $allow_request->value == 0) {
+    $variables['allow_request'] = FALSE;
+    return;
+  }
+
+  if ($group->getMember($account)) {
+    return;
+  }
+
+  $variables['allow_request'] = TRUE;
+  $variables['group_operations_url'] = Url::fromUserInput('/group/' . $group->id() . '/request-membership');
+
+  $contentTypeConfigId = $group
+    ->getGroupType()
+    ->getContentPlugin('group_membership_request')
+    ->getContentTypeConfigId();
+
+  $request = \Drupal::entityTypeManager()->getStorage('group_content')->loadByProperties([
+    'type' => $contentTypeConfigId,
+    'gid' => $group->id(),
+    'entity_id' => $account->id(),
+    'grequest_status' => GroupMembershipRequest::REQUEST_PENDING,
+  ]);
+
+  if (!empty($request)) {
+    $variables['requested'] = TRUE;
+    $variables['group_operations_url'] = Url::fromUserInput('/group/' . $group->id() . '/cancel-request');
+  }
+
+  $variables['#cache']['tags'][] = 'request-membership:' . $group->id();
+}
+
+/**
+ * Implements hook_menu_local_tasks_alter().
+ */
+function social_group_request_menu_local_tasks_alter(&$data, $route_name, RefinableCacheableDependencyInterface &$cacheability) {
+  // Get the Group object from the route.
+  $group = Drupal::routeMatch()->getParameter('group');
+
+  // Must be a group, otherwise we don't do anything.
+  if ($group instanceof Group && isset($data['tabs'][0]['social_group.pending_members'])) {
+    $pending_members_tab = &$data['tabs'][0]['social_group.pending_members'];
+
+    $contentTypeConfigId = $group
+      ->getGroupType()
+      ->getContentPlugin('group_membership_request')
+      ->getContentTypeConfigId();
+
+    $requests = \Drupal::entityTypeManager()->getStorage('group_content')->loadByProperties([
+      'type' => $contentTypeConfigId,
+      'gid' => $group->id(),
+      'grequest_status' => GroupMembershipRequest::REQUEST_PENDING,
+    ]);
+
+    $pending_members_tab['#link']['title'] = sprintf('%s (%s)', $pending_members_tab['#link']['title'], count($requests));
+
+    $cacheability->addCacheTags(['request-membership:' . $group->id()]);
+  }
+
+}
+
+/**
+ * Implements hook_activity_send_email_notifications_alter().
+ */
+function social_group_request_activity_send_email_notifications_alter(array &$items, array $email_message_templates) {
+  if (isset($email_message_templates['request_to_join_a_group'])) {
+    $items['what_manage']['templates'][] = 'request_to_join_a_group';
+  }
+
+  if (isset($email_message_templates['approve_request_join_group'])) {
+    $items['message_to_me']['templates'][] = 'approve_request_join_group';
+  }
+}
+
+/**
+ * Implements hook_form_alter().
+ */
+function social_group_request_form_alter(array &$form, FormStateInterface $form_state, $form_id) {
```

Comment:
```suggestion
function social_group_request_form_group_content_closed_group-group_membership_add_form_alter(array &$form, FormStateInterface $form_state, $form_id) {
```

Reply:
Don't agree with you. On this form_alter I work with 2 different forms

Reply:
You're right.

Reply:
php doesn't allow to create the function with the "-" in the function name.

Also, why you prefer to use separate form_alter hooks instead of using one with if statements?

Reply:
The **hook_form_alter** hook should be used only when need to do something the same for few forms but when need to do something different for different forms then should be used **hook_form_FORM_ID_alter** or **hook_form_BASE_FORM_ID_alter** hooks. Also **if** isn't the best solution for checking form ID because the compiler will be checked conditions in each **if** definition. If you want to check few form IDs in the **hook_form_alter** then better to use **if ... elseif** or **switch** construction.